db/commitlog/commitlog.cc

```diff
@@ -1525,8 +1525,11 @@ future<db::commitlog::segment_manager::sseg_ptr> db::commitlog::segment_manager:
 
         f = make_checked_file(commit_error_handler, std::move(f));
     } catch (...) {
-        ep = std::current_exception();
-        commit_error_handler(ep);
+        try {
+            commit_error_handler(std::current_exception());
+        } catch (...) {
```

Comment:
This throws only in extremely non-default configurations, yes?

Not saying it shouldn't be corrected for those cases, just trying to understand the scope of the problem.

Reply:
This patch is really part of different PR (see top comment), but to answer the question: commit_error_handler rethrows internally, and wraps + rethrows iff system_error, or aborts node iff configured as such. We need to always propagate the error out of catch to do async cleanup. 
It should be very rare to reach this, yes.